src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -369,7 +373,15 @@ internal NamespaceOrTypeOrAliasSymbolWithAnnotations BindNamespaceOrTypeOrAliasS
                         if (ShouldCheckConstraints && constructedType.IsNullableType())
                         {
                             ReportUseSiteDiagnostics(constructedType.TypeSymbol.OriginalDefinition, diagnostics, syntax);
-                            ((NamedTypeSymbol)constructedType.TypeSymbol).CheckConstraints(this.Compilation, this.Conversions, syntax.Location, diagnostics);
+                            var type = (NamedTypeSymbol)constructedType.TypeSymbol;
+                            var location = syntax.Location;
+                            var conversions = this.Conversions.WithNullability(includeNullability);
+                            if (includeNullability && !ShouldCheckConstraintsNullability)
+                            {
+                                diagnostics.Add(new LazyNullableContraintChecksDiagnosticInfo(type, conversions, this.Compilation), location);
+                                conversions = conversions.WithNullability(includeNullability: false);
```

Comment:
>WithNullability [](start = 58, length = 15)

Were you able to hit this scenario? The tests you added cover attribute arguments, which would not enter this path.

Reply:
Attribute arguments hit this path. `AttributeArgumentCycle_Nullable` resulted in a `StackOverflowException` without this change.

---
In reply to: [209389682](https://github.com/dotnet/roslyn/pull/29197#discussion_r209389682) [](ancestors = 209389682)